schemas/ephys/Ephys Session Schema.json

```diff
@@ -0,0 +1,264 @@
+		},
+	"project_name": {
+		"type": "string",
+		"title": "Project name"
+	},
+	"project_id": {
+		"type": "string",
+		"title": "Project ID"
+		},
+	"session_type": {
+		"type": "string", 
+		"enum": ["Test", "Foraging A", "Spontaneous", "Foraging B"],
+		"title": "Session type"
+		},
+	"stimulus_protocol_id": {
+		"type": "string",
+		"title": "Stimulus protocol ID"
+		},
+	"rig_id": {
+		"type": "string",
+		"title": "Rig ID",
+		},
+	"probes": {
+		"type": "array",
+		"items": { "$ref": "#/$defs/probe" },
+		"uniqueItems": true,
+		"title": "Probes"
+		},
+	"lasers": {
+		"type": "array",
+		"items": {"$ref": "#/$defs/laser"},
+		"uniqueItems": true,
+		"title": "Lasers"
+		}
+	},
+	"$defs": {
+		"probe": {
+			"type": "object",
+			"properties": {
+				"name": {
+					"type": "string",
+					"title": "Name"
+					},
+				"tip_targeted_structure": {
+					"type": "string"
+					},
+                "other_targeted_structures": {
+                    "type": "string"
+                },
+                "targeted_ccf_coordinates": {
+                    "type": "array",
+                    "items": {"$ref": "#/$defs/ccf_coords"},
+                    "uniqueItems": true,
+                    "minItems": 3,
+                    "maxItems": 3,
+                    "title": "Targeted CCF coordinates"
+                },
+                "targeted_lab_coordinates": {
+                    "type": "array",
+                    "items": {"$ref": "#/$defs/lab_coords"},
+                    "uniqueItems": true,
+                    "minItems": 3,
+                    "maxItems": 3,
+                    "title": "Targeted lab coordinates",
+                    "description": "Targeted coordinates relative to the headframe"
+                },
+                "manipulator_coordinates": {
+                    "type": "array",
+                    "items": {"$ref": "#/$defs/manipulator_coords"},
+                    "uniqueItems": true,
+                    "minItems": 3,
+                    "maxItems": 3,
+                    "title": "Manipulator coordinates"
+                },
+                "manipulator_angles": {
+                    "type": "array",
+                    "items": {"$ref": "#/$defs/manipulator_angles"},
+                    "uniqueItems": true,
+                    "title": "Manipulator angles"
+                }
+            },
+            "$defs": {
+                "ccf_coords": {
+                    "type": "object",
+                    "properties": {
+                        "direction": {
+                            "type": "string",
+                            "enum": ["ML", "AP", "DV"],
+                            "title": "Direction"
+                        },
+                        "value": {
+                            "type": "number",
+                            "title": "Value"
+                        }
+                    },
+                    "required": ["direction", "value"]
+                },
```

Comment:
lab and manipulator coordinate systems are specified the same way, so I thing we can collapse the subschemas here into one.  call it `coordinates_mm`?

Reply:
But the coordinates are in different reference frames.

Reply:
that's cool they can have different values but share the same schema.

Reply:
e.g.: 
```
{
  'properties': {
    'bob': { 'type': { '$ref': '/defs/person' } },
    'alice': { 'type': { '$ref': '/defs/person' } },
  },
  'defs': {
    'person': {
    ...
    }
 }
}
```
This is syntax isn't right, but hopefully close enough to explain.

Reply:
oh, I see what you mean!